### PR TITLE
revamped snake tutorial

### PR DESCRIPTION
**Pull request by conejoninja**
I'd like @doniacld to take a look too (she doesn't appear on the available reviewers list :'( )

I DID NOT compile nor test the code.

Needs images on the README.md too

Probably needs multiple fixes

### CODE REVIEW

File: tutorial/snake/step7/main.go [hunk 1]
```diff
@@ -169,10 +171,6 @@ func drawSnakePartial(x, y int16, c color.RGBA) {
 func createApple() {
 	appleX = int16(rand.Int31n(WIDTHBLOCKS))
 	appleY = int16(rand.Int31n(HEIGHTBLOCKS))
-	for collisionWithSnake(appleX, appleY) {
```
Comment (conejoninja) [inline tutorial/snake/step7/main.go]: why is this removed?

Comment (deadprogram) [reply tutorial/snake/step7/main.go]: Let me look that might be unintentional.

Comment (deadprogram) [reply tutorial/snake/step7/main.go]: I see now how that was necessary for step7, so now added back in and works as expected. Thanks for pointing that out @conejoninja
